src/main/java/binnie/core/craftgui/minecraft/GuiCraftGUI.java

```diff
@@ -332,86 +338,84 @@ public void renderTexturePadded(IArea area, IArea texture, IBorder padding) {
 		}
 	}
 
-	public void drawGradientArea(float p_73733_1_, float p_73733_2_, float p_73733_3_, float p_73733_4_, int p_73733_5_, int p_73733_6_) {
-		float f = (p_73733_5_ >> 24 & 0xFF) / 255.0f;
-		float f2 = (p_73733_5_ >> 16 & 0xFF) / 255.0f;
-		float f3 = (p_73733_5_ >> 8 & 0xFF) / 255.0f;
-		float f4 = (p_73733_5_ & 0xFF) / 255.0f;
-		float f5 = (p_73733_6_ >> 24 & 0xFF) / 255.0f;
-		float f6 = (p_73733_6_ >> 16 & 0xFF) / 255.0f;
-		float f7 = (p_73733_6_ >> 8 & 0xFF) / 255.0f;
-		float f8 = (p_73733_6_ & 0xFF) / 255.0f;
-		GL11.glDisable(3553);
-		GL11.glEnable(3042);
-		GL11.glDisable(3008);
-		OpenGlHelper.glBlendFunc(770, 771, 1, 0);
-		GL11.glShadeModel(7425);
+	public void drawGradientArea(float x, float y, float width, float height, int color1, int color2) {
+		float alpha1 = (color1 >> 24 & 0xFF) / 255.0f;
+		float r1 = (color1 >> 16 & 0xFF) / 255.0f;
+		float b1 = (color1 >> 8 & 0xFF) / 255.0f;
+		float g1 = (color1 & 0xFF) / 255.0f;
+
+		float alpha2 = (color2 >> 24 & 0xFF) / 255.0f;
+		float r2 = (color2 >> 16 & 0xFF) / 255.0f;
+		float g2 = (color2 >> 8 & 0xFF) / 255.0f;
+		float b2 = (color2 & 0xFF) / 255.0f;
+
+		GL11.glDisable(GL11.GL_TEXTURE_2D);
+		GL11.glDisable(GL11.GL_ALPHA_TEST);
+		GL11.glEnable(GL11.GL_BLEND);
+		OpenGlHelper.glBlendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA, 1, 0);
+		GL11.glShadeModel(GL11.GL_SMOOTH);
 		Tessellator tessellator = Tessellator.instance;
 		tessellator.startDrawingQuads();
-		tessellator.setColorRGBA_F(f2, f3, f4, f);
-		tessellator.addVertex(p_73733_3_, p_73733_2_, zLevel);
-		tessellator.addVertex(p_73733_1_, p_73733_2_, zLevel);
-		tessellator.setColorRGBA_F(f6, f7, f8, f5);
-		tessellator.addVertex(p_73733_1_, p_73733_4_, zLevel);
-		tessellator.addVertex(p_73733_3_, p_73733_4_, zLevel);
+		tessellator.setColorRGBA_F(r1, b1, g1, alpha1);
+		tessellator.addVertex(width, y, zLevel);
+		tessellator.addVertex(x, y, zLevel);
+		tessellator.setColorRGBA_F(r2, g2, b2, alpha2);
+		tessellator.addVertex(x, height, zLevel);
+		tessellator.addVertex(width, height, zLevel);
 		tessellator.draw();
-		GL11.glShadeModel(7424);
-		GL11.glDisable(3042);
-		GL11.glEnable(3008);
-		GL11.glEnable(3553);
+		GL11.glShadeModel(GL11.GL_FLAT);
+		GL11.glDisable(GL11.GL_BLEND);
+		GL11.glEnable(GL11.GL_ALPHA_TEST);
+		GL11.glEnable(GL11.GL_TEXTURE_2D);
 	}
 
 	public void renderItem(IPoint pos, ItemStack item, boolean rotating) {
-		renderItem(pos, item, (int) zLevel + 3, rotating);
-	}
-
-	private void renderItem(IPoint pos, ItemStack item, int zLevel, boolean rotating) {
 		OpenGlHelper.setLightmapTextureCoords(OpenGlHelper.lightmapTexUnit, 240.0f, 240.0f);
-		GL11.glPushMatrix();
+		GL11.glPushAttrib(8256);
```

Comment:
```
public static void pushAttrib()
    {
        GL11.glPushAttrib(8256);
    }
```
in GlStateManager from Forge 1.11.2.
https://github.com/ForestryMC/Binnie/blob/master-MC1.11.2/src/main/java/binnie/core/craftgui/renderer/RenderUtil.java#L45